### ISSUE TITLE
conda and travis

### ISSUE THREAD

**Issue by tpaviot** [tpaviot/pythonocc-contrib#21]
@rainman110 I've been trying to use your conda package for oce/linux on travis, thus speeding up the compilation. The branch I'm working on is review/travis-conda (https://github.com/tpaviot/pythonocc-core/tree/review/travis-conda).

The download/install seems ok, however I get a compilation error I never faced before:

``` bash
Scanning dependencies of target _AIS
[  0%] Building CXX object CMakeFiles/_AIS.dir/src/SWIG_files/wrapper/AISPYTHON_wrap.cxx.o
make[2]: *** No rule to make target `/usr/lib64/libSM.so', needed by `Unix/x86_64-MinSizeRel-64/_AIS.so'.  Stop.
make[1]: *** [CMakeFiles/_AIS.dir/all] Error 2
make: *** [all] Error 2
```

Are you sure you included all the dependencies in your linux package ?

FYI, you can have the full trace at https://travis-ci.org/tpaviot/pythonocc-core/jobs/98849183

**Comment by jf---** [comment]
@tpaviot , looks like this is a missing lib [that can be apt-get'd?](http://askubuntu.com/questions/611933/missing-libsm-so-6-library-qt-creator-installation-error)

**Comment by rainman110** [comment]
@tpaviot , @jf---  It's not a missing lib, as the module ais is loading successfully.

I suppose the problem is the hard-coded path /usr/lib64/libSM.so, which is typical on RPM based distros. I build the package on SLES 11 SP2.

The problem is probably the OCE cmake library export module or a configuration of cmake. In OCE-libraries-release.cmake, there are lots of these hard coded paths. We could either remove them from the config script, or change some cmake option, that system libs are used without absoute paths.

**Comment by jf---** [comment]
+1 for no absolute paths... this is a huge plus...
@rainman110 , could we avoid linking to libSM.so when building?

**Comment by rainman110** [comment]
@jf--- No, I don't think we can avoid linking libSM as it is part of the X11 libs. Each linux distribution should have it, so no problem here.

Actually, there is no need to re-link all these libraries (OpenGL, X11 ...), as they are already linked by oce. It would only be necessary, if we built static OCE libs, which we didn't do.

**TLDR**: We can and should remove all explicit linkings to system libraries in  `OCE-libraries-release.cmake`. These are not required and prohibit a relocatable package. We could automatically remove these entries with a small python script. I am doing this already for one of the DLR-SC libs on windows.

**Comment by jf---** [comment]
Thanks @rainman110 , that's very clear... I didnt know about libSM
yes all for relocatable binaries... 
:+1: for removing the links through a script...

**Comment by rainman110** [comment]
@tpaviot  I wrote a small script that hopefully fixes your problems (DLR-SC/tigl-conda@38efe148b31e8)

I also rebuilt OCE and uploaded it to anaconda/DLR-SC.

Please give me feedback, if this fixes this issue.

**Comment by tpaviot** [comment]
Great, that worked !

**Comment by rainman110** [comment]
Great! So I suppose we can close this issue?

I will back-port my fix to this repository with a pull request.

**Comment by tpaviot** [comment]
I prefer waiting that the travis/conda compilation is fully ok. So far, I get the compilation working, but  I fail to run the unittest (see for instance https://travis-ci.org/tpaviot/pythonocc-core/jobs/99059382). I don't understand why python can't load shared libraries, there must be a conflict somewhere.

**Comment by rainman110** [comment]
It looks like you are mixing python versions.

**Comment by rainman110** [comment]
According to [this link](https://www.mail-archive.com/search?l=python-dev@python.org&q=subject:%22Re%5C%3A+%5C%5BPython%5C-Dev%5C%5D+%5C%5BPython%5C-3000%5C%5D+inst_persistent_id%22&o=newest&f=1), the function `PyInstance_Type`is python2 only. 

So it seems that you've built pythonocc against python2 and you try to run it with python3.

**Comment by tpaviot** [comment]
ok, everything is working now. see https://travis-ci.org/tpaviot/pythonocc-core/builds/99212260

this ticket can be closed.
